### PR TITLE
Expose Building method to get a new multiprocessing pool

### PR DESCRIPTION
**Pull request by dschuff**
Use it for Building.read_link_inputs to work around #4941

### CODE REVIEW

**Comment by dschuff** [comment]
@kripken here's what I think is a bit better (hopefully temporrary) workaround for #4941. WDYT?

File: tools/shared.py [hunk 1]
```diff
@@ -1244,41 +1244,50 @@ class Building:
...
-        class FakeMultiprocessor:
-          def map(self, func, tasks):
-            results = []
-            for t in tasks:
-              results += [func(t)]
-            return results
-        Building.multiprocessing_pool = FakeMultiprocessor()
-      else:
-        child_env = [
+      pool = Building.get_new_multiprocessing_pool()
```
Comment (kripken) [inline tools/shared.py]: why the temp var?

**Comment by kripken** [comment]
I think I don't understand what this does. Does it always create a new pool every time?

**Comment by kripken** [comment]
(in that `read_link_inputs` method)

Without understanding what's going on, I worry about working around it by not reusing the same pool in one method. Why do we think that method is special? Or maybe I'm just confused here.

File: tools/shared.py [hunk 2]
```diff
@@ -1244,41 +1244,50 @@ class Building:
...
+      Building.multiprocessing_pool = pool
+    return Building.multiprocessing_pool
+
+  @staticmethod
+  def get_new_multiprocessing_pool():
```
Comment (jgravelle-google) [inline tools/shared.py]: Naming nit: `get` and `new` are different verbs. Maybe `create_multiprocessing_pool()`?

File: tools/shared.py (hunk 1 above)
Comment (dschuff) [inline tools/shared.py]: Vestigial from a previous revision, will remove.

File: tools/shared.py (hunk 2 above)
Comment (dschuff) [inline tools/shared.py]: Grammar nit: `new` is an adjective, not a verb. I still do like "create" better though.

**Comment by kripken** [comment]
Got it, thanks. Overall this seems good. To make sure I understand, the benefits of this over the previous workaround PR are that we always use a pool, we just create a new one (so just suffer the creation overhead but still get parallelism), and there is no special-casing of the test suite? If I got that right then lgtm.

**Comment by dschuff** [comment]
Yes, exactly.

**Comment by dschuff** [comment]
OK, looks good locally and it looks like the failures on the emscripten waterfall are the same as on wasm-stat.us so hopefully they will go green.

**Comment by juj** [comment]
I have a proper fix that resolves the root issue, so let me revert this as it tackles a symptom rather than the cause.

**Pull request merged**